src/extractor/guidance/coordinate_extractor.cpp

```diff
@@ -261,7 +261,8 @@ CoordinateExtractor::GetCoordinateAlongRoad(const NodeID intersection_node,
         const double offset = 0.5 * considered_lanes * ASSUMED_LANE_WIDTH;
         coordinates = TrimCoordinatesToLength(std::move(coordinates), offset, segment_distances);
         segment_distances.resize(coordinates.size());
-        segment_distances.back() = offset;
+        segment_distances.back() = util::coordinate_calculation::haversineDistance(
```

Comment:
An assertion that `coordinates.size() >= 2` would help document assumptions here.